src/devices/Media/VideoDevice/Devices/UnixVideoDevice.cs

```diff
@@ -0,0 +1,546 @@
+            v4l2_requestbuffers req = new v4l2_requestbuffers
+            {
+                count = BufferCount,
+                type = v4l2_buf_type.V4L2_BUF_TYPE_VIDEO_CAPTURE,
+                memory = v4l2_memory.V4L2_MEMORY_MMAP
+            };
+            V4l2Struct(VideoSettings.VIDIOC_REQBUFS, ref req);
+
+            // Mapping the applied buffer to user space
+            V4l2FrameBuffer[] buffers = new V4l2FrameBuffer[BufferCount];
+            for (uint i = 0; i < BufferCount; i++)
+            {
+                v4l2_buffer buffer = new v4l2_buffer
+                {
+                    index = i,
+                    type = v4l2_buf_type.V4L2_BUF_TYPE_VIDEO_CAPTURE,
+                    memory = v4l2_memory.V4L2_MEMORY_MMAP
+                };
+                V4l2Struct(VideoSettings.VIDIOC_QUERYBUF, ref buffer);
+
+                buffers[i].Length = buffer.length;
+                buffers[i].Start = Interop.mmap(IntPtr.Zero, (int)buffer.length, MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE, MemoryMappedFlags.MAP_SHARED, _deviceFileDescriptor, (int)buffer.m.offset);
+            }
+
+            // Put the buffer in the processing queue
+            for (uint i = 0; i < BufferCount; i++)
+            {
+                v4l2_buffer buffer = new v4l2_buffer
+                {
+                    index = i,
+                    type = v4l2_buf_type.V4L2_BUF_TYPE_VIDEO_CAPTURE,
+                    memory = v4l2_memory.V4L2_MEMORY_MMAP
+                };
+                V4l2Struct(VideoSettings.VIDIOC_QBUF, ref buffer);
+            }
+
+            return buffers;
+        }
+
+        private static unsafe void UnmappingFrameBuffers(V4l2FrameBuffer* buffers)
+        {
+            // Unmapping the applied buffer to user space
+            for (uint i = 0; i < BufferCount; i++)
+            {
+                Interop.munmap(buffers[i].Start, (int)buffers[i].Length);
+            }
+        }
+
+        private unsafe void SetVideoConnectionSettings()
+        {
+            FillVideoConnectionSettings();
+            // Set capture format
+            v4l2_format format = new v4l2_format
+            {
+                type = v4l2_buf_type.V4L2_BUF_TYPE_VIDEO_CAPTURE,
+                fmt = new fmt
+                {
+                    pix = new v4l2_pix_format
+                    {
+                        width = Settings.CaptureSize.Width,
+                        height = Settings.CaptureSize.Height,
+                        pixelformat = Settings.PixelFormat
+                    }
+                }
+            };
+            var res = V4l2Struct(VideoSettings.VIDIOC_S_FMT, ref format);
+
+            // Set exposure type
+            v4l2_control ctrl = new v4l2_control
+            {
+                id = VideoDeviceValueType.ExposureType,
+                value = (int)Settings.ExposureType
+            };
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
```

Comment:
I feel like V4l2Struct should have an overload which take id and value so that you don't have to touch the struct directly

Reply:
This method can operate on all v4l2 structs. When setting these settings, they just used a struct together. I think that's fine now.🤔